devtools/clean_topo.h

```diff
@@ -0,0 +1,13 @@
+#ifndef LIGHTNING_DEVTOOLS_CLEAN_TOPO_H
+#define LIGHTNING_DEVTOOLS_CLEAN_TOPO_H
+#include "config.h"
+
+struct gossmap;
+
+/* Cleans topology:
+ * 1. Removes channels not enabled in both dirs.
+ * 2. (if remove_singles) remove nodes with only one connection.
+ * 3. Remove isolated nodes (we assume first node is well-connected!).
```

Comment:
This is generally called a strongly connected component of the graph (or connected component in an undirected graph) :-)